src/common/common.c

```diff
@@ -228,6 +228,10 @@ OQS_API int OQS_CPU_has_extension(OQS_CPU_EXT ext) {
 }
 
 OQS_API void OQS_init(void) {
+#if defined(OQS_USE_OPENSSL)
+	OSSL_FUNC(OPENSSL_init_crypto)(OPENSSL_INIT_ENGINE_ALL_BUILTIN | OPENSSL_INIT_ADD_ALL_CIPHERS,
+	                               NULL);
```

Comment:
Thanks for the proposal @ashman-p -- but allow me to ask why an additional init call would fix some apparently missing memory cleanup (at exit)? Or can you explain the choice of parameters here that would facilitate that? From the [documentation of those](https://docs.openssl.org/1.1.1/man3/OPENSSL_init_crypto/#description) it's not quite obvious to me (why these have been chosen and how they contribute to fixing the issue):

> OPENSSL_INIT_ENGINE_ALL_BUILTIN

> With this option the library will automatically load and initialise all the built in engines listed above with the exception of the openssl and afalg engines. This not a default option.

> OPENSSL_INIT_ADD_ALL_CIPHERS

> With this option the library will automatically load and make available all libcrypto ciphers. This option is a default option. Once selected subsequent calls to OPENSSL_init_crypto() with the option OPENSSL_INIT_NO_ADD_ALL_CIPHERS will be ignored.

Reply:
@baentsch, memory leak test failures occurred in test_kem, test_signatures, ... these options seemed to address all the failures. Neil's and Spencer's findings are perhaps pointing more to a root cause.

Reply:
> are perhaps pointing more to a root cause

And (finding and solving) that would be my preference. A workaround imo carries the risk of "getting bitten" in the future again (and/or reduces the opportunity to improve any OSS, be the problem in OQS or OpenSSL). Time permitting of course. Apologies for being anal, @ashman-p . Closing the comment such as to not block a merge.